pkg/cli/cliexecuter.go

```diff
@@ -1,97 +1,263 @@
+			ctxWithTimeout, cancel := context.WithTimeout(ctx, timeout)
+			defer cancel()
+			stdout, stderr, faErr = e.runner(ctxWithTimeout, command)
+			if faErr == nil {
+				e.log.Info("command completed", "uid", uid, "response", stdout, "errMessage", stderr, "err", faErr)
+				return true, nil
+			}
+
+			if wait.Interrupted(faErr) {
+				e.log.Error(faErr, "fence agent timeout", "uid", uid)
+				return false, faErr
+			}
+
+			e.log.Info("command failed", "uid", uid, "response", stdout, "errMessage", stderr, "err", faErr)
+			return false, nil
+		})
+
+	e.log.Info("fence agent done", "uid", uid, "command", command, "stdout", stdout, "stderr", stderr, "err", faErr)
+	return retryErr, faErr
+}
+
+func (e *Executer) updateStatusWithRetry(ctx context.Context, uid types.UID, fenceAgentErr error) error {
+	// Update FAR status with an exponantial backoff retry to handle only the updateStatus error cases where:
+	// - FAR cannot be found, but it does exist
+	// - the status update fails for conflicts
+
+	e.log.Info("updating status", "FAR uid", uid)
+
+	err := wait.ExponentialBackoffWithContext(ctx,
+		retry.DefaultBackoff,
+		func(ctx context.Context) (bool, error) {
+			far, err := e.getFenceAgentsRemediationByUID(ctx, uid)
+			if err != nil {
+				if apiErrors.IsNotFound(err) {
+					e.log.Info("could not find FAR by UID", "FAR uid", uid)
+					return false, err
+				}
+
+				if wait.Interrupted(err) {
+					e.log.Info("could not update status", "FAR uid", uid, "reason", err)
+					return false, err
+				}
+
+				e.log.Error(err, "could not update status", "FAR uid", uid)
+				return false, err
+			}
+
+			if err := e.updateStatus(ctx, far, fenceAgentErr); err != nil {
+				if wait.Interrupted(err) {
+					e.log.Info("context cancelled while updating the status", "FAR uid", uid)
+					return true, err
+				}
+				if apiErrors.IsConflict(err) {
+					e.log.Error(err, "conflict while updating the status", "FAR uid", uid)
+					return true, nil
+				}
+				e.log.Error(err, "failed to update status", "FAR uid", uid)
+				return false, err
+			}
+
+			e.log.Info("status updated", "FAR uid", uid)
+			return true, nil
+		})
+	return err
+}
+
+// Exists checks if there is already a running Fence Agent command mapped to the UID
+func (e *Executer) Exists(uid types.UID) bool {
+	e.routinesLock.Lock()
+	defer e.routinesLock.Unlock()
+	_, exist := e.routines[uid]
+	return exist
+}
+
+func (e *Executer) Remove(uid types.UID) {
+	e.routinesLock.Lock()
+	defer e.routinesLock.Unlock()
+	if routine, exist := e.routines[uid]; exist {
+		e.log.Info("cancelling fence agent routine", "uid", uid)
+		routine.cancel()
+		delete(e.routines, uid)
+	}
+}
+
+// run runs the command in the container and updates the status of the FAR instance maching the UID
+func run(ctx context.Context, command []string) (stdout, stderr string, err error) {
+	cmd := exec.CommandContext(ctx, command[0], command[1:]...)
+
+	var outBuilder, errBuilder strings.Builder
+	cmd.Stdout = &outBuilder
```

Comment:
`cmd.Stdout` and `cmd.Stderr` are `nil` unless defined otherwise.
In this case I want to build a `string` out of them, so I used `string.Builder`.

It's similar to what [CombinedOutput](https://cs.opensource.google/go/go/+/refs/tags/go1.21.5:src/os/exec/exec.go;l=995) does, but with `bytes.Buffer`